web_m2x_options/static/src/components/form.esm.js

```diff
@@ -0,0 +1,380 @@
+    },
+    async loadOptionsSource(request) {
+        var options = await super.loadOptionsSource(request);
+        this.field_color = this.props.fieldColor;
+        this.colors = this.props.fieldColorOptions;
+        if (this.colors && this.field_color) {
+            var value_ids = options.map((result) => result.value);
+            const objects = await this.orm.call(
+                this.props.resModel,
+                "search_read",
+                [],
+                {
+                    domain: [["id", "in", value_ids]],
+                    fields: [this.field_color],
+                }
+            );
+            for (var index in objects) {
+                for (var index_value in options) {
+                    if (options[index_value].value === objects[index].id) {
+                        // Find value in values by comparing ids
+                        var option = options[index_value];
+                        // Find color with field value as key
+                        var color =
+                            this.colors[objects[index][this.field_color]] || "black";
+                        option.style = "color:" + color;
+                        break;
+                    }
+                }
+            }
+        }
+        return options;
+    },
+});
+
+patch(FormController.prototype, {
+    /**
+     * @override
+     */
+    setup() {
+        super.setup(...arguments);
+        this._setSubViewLimit();
+    },
+    /**
+     * @override
+     * add more method to add subview limit on formview
+     */
+    async _setSubViewLimit() {
+        const ir_options = session.web_m2x_options || {};
+        const activeFields = this.archInfo.fieldNodes,
+            isSmall = this.user;
+
+        var limit = ir_options["web_m2x_options.field_limit_entries"];
+        if (!(typeof limit === "undefined")) {
+            limit = parseInt(limit, 10);
+        }
+        for (const fieldName in activeFields) {
+            const field = activeFields[fieldName];
+            if (!isX2Many(field)) {
+                // What follows only concerns x2many fields
+                continue;
+            }
+            // Const fieldInfo = activeFields[fieldName];
+            if (field.invisible) {
+                // No need to fetch the sub view if the field is always invisible
+                continue;
+            }
+
+            if (!field.field.useSubView) {
+                // The FieldComponent used to render the field doesn't need a sub view
+                continue;
+            }
+            let viewType = field.viewMode || "list,kanban";
+            viewType = viewType.replace("tree", "list");
+            if (viewType.includes(",")) {
+                viewType = isSmall ? "kanban" : "list";
+            }
+            field.viewMode = viewType;
+            if (field.views && field.views[viewType] && limit) {
+                field.views[viewType].limit = limit;
+            }
+        }
+    },
+});
```

Comment:
Thank you for opening the PR!

I notice that O.W.L. throws a blocking Error once you install some module that can benefit from this OCA module (such as `sale_management`) and you enter developer mode. It's because v18's dev mode now validates the object that is added to the registry and notices that the functions below aren't declared in the schema. I'd propose to either incorporate the functions inside `extractProps` (which _is_ declared in the schema) or to extend the `validationSchema` as described below (and add an `import {registry} from "@web/core/registry";`).

(If someone has a better suggestion, I encourage you to correct me, though.)

The error unfortunately also occurs in the PR of @chaule97. If you incorporate their changes, keep in mind that they renamed the functions to JS standard. 

```suggestion
});

// O.W.L. v18+: schema is validated in dev mode on adding to registry
patch(registry.category("fields").validationSchema, {
    m2o_options_props_create: { type: Function, optional: true },
    m2o_options_props_create_edit: { type: Function, optional: true },
    m2o_options_props_limit: { type: Function, optional: true },
    m2o_options_props_search_more: { type: Function, optional: true },
    m2o_options_props_open: { type: Function, optional: true },
    m2o_options_props: { type: Function, optional: true },
    m2m_options_props_create: { type: Function, optional: true },
    m2m_options_props_create_edit: { type: Function, optional: true },
    m2m_options_props_limit: { type: Function, optional: true },
    m2m_options_props_search_more: { type: Function, optional: true },
    m2m_options_props: { type: Function, optional: true },
});
```